src/library/ksLegend/ksLegend.jsx

```diff
@@ -1,19 +1,37 @@
 import React from 'react';
-
 import './ksLegend.scss';
 
+const LEGEND_LIMIT = 5;
+
+const LegendItems = ({ colors, data }) => {
+  return (
+    <>
+      {data.slice(0, LEGEND_LIMIT).map((legend, i) => (
+        <div key={legend.label} className="legend-item">
+          <div className="marker" style={{ backgroundColor: colors && colors[i] }}></div>
+          {legend.label}
+        </div>
+      ))}
+    </>
+  );
+};
+
 const KsLegend = ({ colors, data }) => {
-  const LEGEND_LIMIT = 5;
-  const legendItems = data.slice(0,LEGEND_LIMIT).map((legend, i) =>
-    <div key={legend.label} className="legend-item">
-      <div className="marker" style={{backgroundColor: colors[i]}}></div>
-      {legend.label}
+  if (!data) return null;
```

Comment:
Maybe you can use `isArray` instead?
```suggestion
  if (!Array.isArray(data)) return null;
```